AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -265,13 +270,15 @@ - (void)deleteDocumentWithPartition:(NSString *)partition
                        writeOptions:(MSWriteOptions *)__unused writeOptions
                   completionHandler:(MSDataSourceErrorCompletionHandler)completionHandler {
   // TODO consume writeOptions
-  [MSTokenExchange tokenAsync:(MSStorageIngestion *)self.ingestion
-                   partitions:@[ partition ]
-            completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull error) {
+    [MSTokenExchange
+     performDbTokenAsyncOperationWithHttpClient:(MSStorageIngestion *)self.ingestion
+     partition:partition
+            completionHandler:^(MSTokensResponse *_Nonnull tokenResponses, NSError *_Nonnull tokenExchangeError) {
               // If error getting token.
-              if (error || !tokenResponses) {
-                MSLogError([MSDataStore logTag], @"Can't get CosmosDb token:%@", [error description]);
-                completionHandler([[MSDataSourceError alloc] initWithError:error]);
+              if (tokenExchangeError || !tokenResponses || [tokenResponses.tokens count] <= 0) {
```

Comment:
nit-picking: In Objective-C, checking property of a `nil` object won't throw any errors, hence `!tokenResponses || [tokenResponses.tokens count] <= 0` can be simplified as `[tokenResponses.tokens count] <= 0`.

super nit-picking: `count` can never return negative number, so `[tokenResponses.tokens count] == 0` is slightly more readable than `[tokenResponses.tokens count] <= 0`.

Reply:
cool, I'll make the change. Out of curiosity, what does checking the property on a `nil` object return if it doesn't thrown an error?

Reply:
@anjbur checking property of a `nil` object returns `nil`.